x/oracle/client/cli/tx.go

```diff
@@ -0,0 +1,172 @@
+package cli
+
+import (
+	"encoding/base64"
+	"encoding/hex"
+	"encoding/json"
+	"errors"
+	"fmt"
+
+	"github.com/spf13/cobra"
+
+	"github.com/cosmos/cosmos-sdk/client"
+	"github.com/cosmos/cosmos-sdk/client/flags"
+	"github.com/cosmos/cosmos-sdk/client/tx"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdktx "github.com/cosmos/cosmos-sdk/types/tx"
+	"github.com/cosmos/cosmos-sdk/version"
+	authtypes "github.com/cosmos/cosmos-sdk/x/auth/types"
+	govcli "github.com/cosmos/cosmos-sdk/x/gov/client/cli"
+	govtypes "github.com/cosmos/cosmos-sdk/x/gov/types"
+
+	"github.com/provenance-io/provenance/x/oracle/types"
+)
+
+// The flag to specify that the command should be ran as a gov proposal
+const FlagGovProposal = "gov-proposal"
+
+// NewTxCmd is the top-level command for oracle CLI transactions.
+func NewTxCmd() *cobra.Command {
+	txCmd := &cobra.Command{
+		Use:                        types.ModuleName,
+		Aliases:                    []string{"t"},
+		Short:                      "Transaction commands for the oracle module",
+		DisableFlagParsing:         true,
+		SuggestionsMinimumDistance: 2,
+		RunE:                       client.ValidateCmd,
+	}
+
+	txCmd.AddCommand(
+		GetCmdSendQuery(),
+		GetCmdOracleUpdate(),
+	)
+
+	return txCmd
+}
+
+// GetCmdOracleUpdate is a command to update the address of the module's oracle
+func GetCmdOracleUpdate() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:     "update <address>",
+		Short:   "Update the module's oracle address",
+		Long:    "Submit an update oracle via governance proposal along with an initial deposit.",
+		Args:    cobra.ExactArgs(1),
+		Aliases: []string{"u"},
+		Example: fmt.Sprintf(`%[1]s tx oracle update pb1skjwj5whet0lpe65qaq4rpq03hjxlwd9nf39lk --deposit 50000nhash`, version.AppName),
+		RunE: func(cmd *cobra.Command, args []string) (err error) {
+			clientCtx, err := client.GetClientTxContext(cmd)
+			if err != nil {
+				return err
+			}
+
+			authority := authtypes.NewModuleAddress(govtypes.ModuleName)
+
+			msg := types.NewMsgUpdateOracle(
+				authority.String(),
+				args[0],
+			)
+
+			proposal, govErr := govcli.ReadGovPropFlags(clientCtx, cmd.Flags())
+			if govErr != nil {
+				return govErr
+			}
+			anys, govErr := sdktx.SetMsgs([]sdk.Msg{msg})
+			if govErr != nil {
+				return govErr
+			}
+			proposal.Messages = anys
+			var req sdk.Msg = proposal
+
+			if err := msg.ValidateBasic(); err != nil {
+				return err
+			}
+			return tx.GenerateOrBroadcastTxCLI(clientCtx, cmd.Flags(), req)
```

Comment:
Since you've already got the `govErr` variable created, you can just assign directly to `proposal.Messages`. Also, I don't think you need to create the `req` variable since the proposal is already an `sdk.Msg` and can be provided to  GenerateOrBroadcastTxCLI.

```suggestion
			proposal.Messages, govErr = sdktx.SetMsgs([]sdk.Msg{msg})
			if govErr != nil {
				return govErr
			}

			if err := msg.ValidateBasic(); err != nil {
				return err
			}
			return tx.GenerateOrBroadcastTxCLI(clientCtx, cmd.Flags(), proposal)
```